care/facility/api/viewsets/bed.py

```diff
@@ -240,3 +246,42 @@ def get_queryset(self):
             allowed_facilities = get_accessible_facilities(user)
             queryset = queryset.filter(bed__facility__id__in=allowed_facilities)
         return queryset
+
+    @extend_schema(
+        description="Toggle patient privacy",
+        responses={status.HTTP_200_OK: "{'privacy': 'boolean'}"},
+        request=None,
+        tags=["consultationbed"],
+    )
+    @action(detail=True, methods=["PATCH"])
+    def toggle_patient_privacy(self, request, external_id):
+        user: User = request.user
+        consultation_bed: ConsultationBed = self.get_queryset().get(
+            external_id=external_id
+        )
```

Comment:
Were you not able to simply do `self.get_object()` since this is a `detail=True` view.

```suggestion
        instance = self.get_object()
```